packages/broker/src/plugins/operator/MaintainTopologyHelper.ts

```diff
@@ -3,108 +3,107 @@ import { Provider } from "@ethersproject/providers"
-        this.logger.info(`OperatorClient created for ${config.operatorContractAddress}`)
     }
 
     async start(): Promise<void> {
-        this.logger.info("Starting OperatorClient")
-        this.logger.info("Subscribing to Staked and Unstaked events")
-        const latestBlock = await this.contract.provider.getBlockNumber()
-        this.contract.on("Staked", async (sponsorship: string) => {
-            this.logger.info(`got Staked event ${sponsorship}`)
-            const sponsorshipAddress = sponsorship.toLowerCase()
-            const streamId = await this.getStreamId(sponsorshipAddress)
+        logger.info('Starting')
+        const latestBlock = await this.operatorContract.provider.getBlockNumber()
+
+        this.operatorContract.on('Staked', async (sponsorship: string) => {
+            logger.info('Receive "Staked" event', { sponsorship })
+            const sponsorshipAddress = toEthereumAddress(sponsorship)
+            const streamId = await this.getStreamId(sponsorshipAddress) // TODO: add catching here
             if (this.streamIdOfSponsorship.has(sponsorshipAddress)) {
-                this.logger.info(`Sponsorship ${sponsorship} already staked into, ignoring`)
+                logger.debug('Ignore already staked into sponsorship', { sponsorship })
                 return
             }
             this.streamIdOfSponsorship.set(sponsorshipAddress, streamId)
 
-            const sponsorshipCount = (this.sponsorshipCountOfStream.get(streamId) || 0) + 1
+            const sponsorshipCount = (this.sponsorshipCountOfStream.get(streamId) ?? 0) + 1
             this.sponsorshipCountOfStream.set(streamId, sponsorshipCount)
             if (sponsorshipCount === 1) {
-                this.emit("addStakedStream", [streamId])
+                this.emit('addStakedStream', [streamId])
             }
         })
-        this.contract.on("Unstaked", async (sponsorship: string) => {
-            this.logger.info(`got Unstaked event ${sponsorship}`)
-            const sponsorshipAddress = sponsorship.toLowerCase()
+        this.operatorContract.on('Unstaked', (sponsorship: string) => {
+            logger.info('Receive "Unstaked" event', { sponsorship })
+            const sponsorshipAddress = toEthereumAddress(sponsorship)
             const streamId = this.streamIdOfSponsorship.get(sponsorshipAddress)
-            if (!streamId) {
-                this.logger.error("Sponsorship not found!")
+            if (streamId === undefined) {
+                logger.debug('Unable to find streamId for sponsorship', { sponsorshipAddress })
                 return
             }
             this.streamIdOfSponsorship.delete(sponsorshipAddress)
-            const sponsorshipCount = (this.sponsorshipCountOfStream.get(streamId) || 1) - 1
+            const sponsorshipCount = (this.sponsorshipCountOfStream.get(streamId) ?? 1) - 1
             this.sponsorshipCountOfStream.set(streamId, sponsorshipCount)
```

Comment:
Is this intentional to have `1` as a default if there are no sponsorships? Or should it be `0` as in line 67?

Reply:
That would make the `sponsorshipCount` negative which would be invalid.

Reply:
Removed defensive programming